src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -441,15 +442,21 @@ def query(self, filters=None, options=None):
                     'type': type_.upper(),
                 })
         else:
+            extra = options.pop('extra', {})
+            state_options = {
+                'snapshots': extra.pop('snapshots', False),
+                'recursive': extra.pop('recursive', True),
+                'snapshots_recursive': extra.pop('snapshots_recursive', False)
+            }
```

Comment:
I dont think we should `pop`, that will change the object from the caller. Possibly irrelevant in this case since its a new object, but still.